mmv1/products/vertexai/IndexEndpointDeployedIndex.yaml

```diff
@@ -0,0 +1,267 @@
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+  include_project: true
+description: |-
+  An endpoint indexes are deployed into. An index endpoint can have multiple deployed indexes.
+autogen_async: false
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "vertex_ai_index_endpoint_deployed_index_basic"
+    primary_resource_id: "basic_deployed_index"
+    vars:
+      endpoint_name: 'endpoint-name'
+      network_name: 'network-name'
+      deployed_index_id: 'deployed_index_id'
+      display_name: "vertex-deployed-index"
+      display_name_index: 'test-index'
+      bucket_name: "bucket-name"
+      address_name: 'vertex-ai-range'
+    test_vars_overrides:
+      network_name: 'acctest.BootstrapSharedServiceNetworkingConnection(t, "vpc-network-1")'
+      address_name: 'acctest.BootstrapSharedTestGlobalAddress(t, "vpc-network-1", acctest.AddressWithPrefixLength(8))'
+  - !ruby/object:Provider::Terraform::Examples
+    name: "vertex_ai_index_endpoint_deployed_index_dedicated_resources"
+    primary_resource_id: "dedicated_resources"
+    vars:
+      display_name_index: 'test-index'
+      bucket_name: "bucket-name"
+      display_name: "vertex-deployed-index"
+      deployed_index_id: 'deployed_index_id'
+    skip_docs: true
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  encoder: templates/terraform/encoders/vertex_ai_index_endpoint_deployed_index.go.erb
+  update_encoder: templates/terraform/update_encoder/vertex_ai_index_endpoint_deployed_index.go.erb
+  decoder: templates/terraform/decoders/vertex_ai_index_endpoint_deployed_index.go.erb
+  pre_delete: templates/terraform/pre_delete/vertex_ai_index_endpoint_deployed_index.go.erb
+  custom_import: templates/terraform/custom_import/vertex_ai_index_endpoint_deployed_index.go.erb
+
+parameters:
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'indexEndpoint'
+    description: |
+      Identifies the index endpoint. Must be in the format
+      'projects/{{project}}/locations/{{region}}/indexEndpoints/{{indexEndpoint}}'
+    required: true
+    immutable: true
+    resource: 'IndexEndpoint'
+    imports: 'name'
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    output: true
+    description: |
+      The name of the DeployedIndex resource.
+  - !ruby/object:Api::Type::String
+    name: 'deployedIndexId'
+    description: The user specified ID of the DeployedIndex. The ID can be up to 128 characters long and must start with a letter and only contain letters, numbers, and underscores. The ID must be unique within the project it is created in.
+    required: true
+  - !ruby/object:Api::Type::String
+    name: 'index'
+    description: The name of the Index this is the deployment of.
+    ignore_read: true
```

Comment:
I saw in [this comment](https://github.com/GoogleCloudPlatform/magic-modules/pull/11039#issuecomment-2209316648) you ignore_read this field because there's a diff where the API returns the project number but the provider wants to set the project id. Have you considered using the [`tpgresource.ProjectNumberDiffSuppress` diff suppress func](https://github.com/GoogleCloudPlatform/magic-modules/blob/a72543a018c089c8579bce785ed34589af22e174/mmv1/third_party/terraform/tpgresource/go/common_diff_suppress.go#L94)?

Reply:
latest commit replaces `ignore_read` eith the `ProjectNumberDiffSuppress` func